### PR TITLE
[networkdirect-sdk] Rework portfile.cmake to allow windows x64 and x86, add fail for other triplet combinations, remove all triplets for CI fail list

### LINKED ISSUE #14687: [networkdirect-sdk] x86-windows x86-windows-static build failure
**Host Environment**
- OS: [Microsoft Windows [Version 10.0.19042.630]]
- Compiler:
   - Visual Studio 2017 Version: 15.9.29  VS Tools Version: 14.16.27023 
     Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27044 for x86
   - Visual Studio 2019 Version: 16.8.2  VS Tools Version: 14.28.29333 
     Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29334 for x86

**To Reproduce**
`vcpkg.exe install networkdirect-sdk:x86-windows networkdirect-sdk:x86-windows-static --keep-going`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
>Computing installation plan...
>The following packages will be built and installed:
>    networkdirect-sdk[core]:x86-windows
>    networkdirect-sdk[core]:x86-windows-static
>Detecting compiler hash for triplet x86-windows...
>Detecting compiler hash for triplet x86-windows-static...
>Could not locate cached archive: >C:\Users\Tim.Autry\AppData\Local\vcpkg\archives\1c\1c7728eac62fc916e6f0ca85200569e82a46a1f3.zip
>Could not locate cached archive: >C:\Users\Tim.Autry\AppData\Local\vcpkg\archives\15\159b09ed842dae12e74b09dad014e9e8e0f812b4.zip
>Starting package 1/2: networkdirect-sdk:x86-windows
>Building package networkdirect-sdk[core]:x86-windows...
>CMake Error at ports/networkdirect-sdk/portfile.cmake:7 (message):
**>networkDirect-sdk only supports x64 and x86**
>Call Stack (most recent call first):
>  scripts/ports.cmake:136 (include)
>
>Error: Building package networkdirect-sdk:x86-windows failed with: BUILD_FAILED
>Elapsed time for package networkdirect-sdk:x86-windows: 42.06 ms
>Starting package 2/2: networkdirect-sdk:x86-windows-static
>Building package networkdirect-sdk[core]:x86-windows-static...
>-- Using community triplet x86-windows-static. This triplet configuration is not guaranteed to succeed.
>-- [COMMUNITY] Loading triplet configuration from: D:\Git\vcpkg\vs2017-x86-v141\triplets\community\x86-windows-static.cmake
>CMake Error at ports/networkdirect-sdk/portfile.cmake:7 (message):
**>networkDirect-sdk only supports x64 and x86**
>Call Stack (most recent call first):
>  scripts/ports.cmake:136 (include)
>
>Error: Building package networkdirect-sdk:x86-windows-static failed with: BUILD_FAILED
>Elapsed time for package networkdirect-sdk:x86-windows-static: 42.79 ms
>
>Total elapsed time: 2.966 s
>
>RESULTS
>    networkdirect-sdk:x86-windows: BUILD_FAILED: 42.06 ms
>    networkdirect-sdk:x86-windows-static: BUILD_FAILED: 42.79 ms
>
>SUMMARY
>    SUCCEEDED: 0
>    BUILD_FAILED: 2
>    POST_BUILD_CHECKS_FAILED: 0
>   FILE_CONFLICTS: 0
>    CASCADED_DUE_TO_MISSING_DEPENDENCIES: 0
>    EXCLUDED: 0
>

- Please attach any additional failure logs mentioned in the console output.
No other logs

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
- triplets x64-windows and x64-windows-static NOT effected

- Reference file `[vcpkg-root]\ports\networkdirect-sdk\portfile.cmake` https://github.com/microsoft/vcpkg/blob/42456b785a4508386f080ae8fad2642e056b747d/ports/networkdirect-sdk/portfile.cmake#L6
the `"x64" OR` should be `"x64" AND NOT`

### PR DESCRIPTION
**Pull request by StarGate-One**
**Describe the pull request**

- What does your PR fix? Fixes #14687 [networkdirect-sdk] x86-windows x86-windows-static build failure 

- Which triplets are supported/not supported? 
   - Supported: x86-windows x86-windows-static(community triplet) x64-windows x64-windows-static and NOT UWP
   - Not supported: Those not mentioned above
   - Have you updated the CI baseline? Removed **networkdirect-sdk:[triplets]** from CI fail as it is handled within the `portfile.cmake` now.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/networkdirect-sdk/portfile.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ if (VCPKG_CMAKE_SYSTEM_NAME)
     message(FATAL_ERROR "networkDirect-sdk only supports windows")
 endif()
```
Comment (NancyLi1013) [inline ports/networkdirect-sdk/portfile.cmake:4]: Could you please update the unsupported message as `vcpkg_fail_port_install(ON_TARGET "linux" "osx")`?

File: ports/networkdirect-sdk/portfile.cmake [hunk 2]
```diff
@@ -3,7 +3,7 @@ if (VCPKG_CMAKE_SYSTEM_NAME)
     message(FATAL_ERROR "networkDirect-sdk only supports windows")
 endif()
 
-if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x64" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x64" AND NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
     message(FATAL_ERROR "networkDirect-sdk only supports x64 and x86")
 endif()
```
Comment (NancyLi1013) [inline ports/networkdirect-sdk/portfile.cmake:8]: I think these codes can be combined with the above codes.

`vcpkg_fail_port_install(ON_ARCH "arm"  ON_TAREGT "linux" "osx")`

File: ports/networkdirect-sdk/portfile.cmake [hunk 3]
```diff
@@ -3,7 +3,7 @@ if (VCPKG_CMAKE_SYSTEM_NAME)
...
```
Comment (NancyLi1013) [inline ports/networkdirect-sdk/portfile.cmake:9]: Could you please update 

`file(COPY ${LICENSE} DESTINATION ${CURRENT_PACKAGES_DIR}/share/networkdirect-sdk/COPYRIGHT)`

as

`file(INSTALL ${LICENSE} DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}  RENAME copyright)`
?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by StarGate-One** [comment]
1. Closed - Some of us do not have time to rewrite ports to meet the ever changing undocumented standards, especially since most of us are community volunteers.
2. No hard feelings, just do not have the time after the issue with CURL this past Friday.

File: ports/networkdirect-sdk/portfile.cmake (hunk 2 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake:8]: I add uwp also...

File: ports/networkdirect-sdk/portfile.cmake (hunk 3 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake:9]: Ok

File: ports/networkdirect-sdk/portfile.cmake (hunk 1 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake:4]: Ok

File: ports/networkdirect-sdk/portfile.cmake [hunk 4]
```diff
@@ -1,10 +1,10 @@
 #only supports windows
 if (VCPKG_CMAKE_SYSTEM_NAME)
```
Comment (NancyLi1013) [inline ports/networkdirect-sdk/portfile.cmake:2]:
```suggestion
```

File: ports/networkdirect-sdk/portfile.cmake [hunk 5]
```diff
@@ -1,10 +1,10 @@
 #only supports windows
 if (VCPKG_CMAKE_SYSTEM_NAME)
-    message(FATAL_ERROR "networkDirect-sdk only supports windows")
+    vcpkg_fail_port_install(ON_ARCH "arm" "uwp" ON_TAREGT "linux" "osx")
 endif()
```
Comment (NancyLi1013) [inline ports/networkdirect-sdk/portfile.cmake:4]:
```suggestion
    vcpkg_fail_port_install(ON_ARCH "arm"  ON_TAREGT "uwp" "linux" "osx")
```

File: ports/networkdirect-sdk/portfile.cmake (hunk 2 above)
Comment (NancyLi1013) [reply ports/networkdirect-sdk/portfile.cmake:8]: These codes are not needed here. Since `vcpkg_fail_port_install()` has included this part.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/networkdirect-sdk/portfile.cmake (hunk 4 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake]: Removed if (VCPKG_CMAKE_SYSTEM_NAME) and endif() - lines 1 and 3, fixed formatting by removing indentation on line 2

File: ports/networkdirect-sdk/portfile.cmake (hunk 5 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake]: Moved "uwp" from ON_ARCH to ON_TARGET

**Review by NancyLi1013 — APPROVED** [review]

File: ports/networkdirect-sdk/portfile.cmake [hunk 6]
```diff
@@ -1,11 +1,5 @@
 #only supports windows
-if (VCPKG_CMAKE_SYSTEM_NAME)
-    message(FATAL_ERROR "networkDirect-sdk only supports windows")
-endif()
-
-if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x64" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-    message(FATAL_ERROR "networkDirect-sdk only supports x64 and x86")
-endif()
+vcpkg_fail_port_install(ON_ARCH "arm" ON_TAREGT "uwp" "linux" "osx")
```
Comment (strega-nil) [inline ports/networkdirect-sdk/portfile.cmake:2]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp" "linux" "osx")

```

File: ports/networkdirect-sdk/CONTROL [hunk 7]
```diff
@@ -1,5 +1,6 @@
 Source: networkdirect-sdk
 Version: 2.0.1
+Port-Version: 1
 Description: The Network Direct architecture allows hardware vendors to expose the advanced capabilities of their networking devices.
 Homepage: https://www.nuget.org/packages/NetworkDirect
 Supports: windows&(x64|x86)
```
Comment (strega-nil) [inline ports/networkdirect-sdk/CONTROL:6]:
```suggestion
Supports: windows & (x64|x86) & !uwp
```

File: ports/networkdirect-sdk/portfile.cmake (hunk 6 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/portfile.cmake]: Fixed.

File: ports/networkdirect-sdk/CONTROL (hunk 7 above)
Comment (StarGate-One) [reply ports/networkdirect-sdk/CONTROL]: Fixed.

**Comment by StarGate-One** [comment]
@NancyLi1013 @strega-nil - thank you for you hard work 👍🏻 
I appreciate all you help and suggestions.

**Comment by strega-nil** [comment]
Awesome, thanks @StarGate-One !

File: ports/networkdirect-sdk/portfile.cmake (hunk 6 above)
Comment (NancyLi1013) [reply ports/networkdirect-sdk/portfile.cmake]: Sorry for this typo error, it's my fault.

**Pull request merged**